common/aws/test/client_test.go

```diff
@@ -184,7 +179,11 @@ func TestReadNonExistentValue(t *testing.T) {
 
 		client, err := builder.build()
 		assert.NoError(t, err)
-		ReadNonExistentValueTest(t, client)
+		_, err = client.FragmentedDownloadObject(context.Background(), bucket, "nonexistent", 1000, 1000)
+		assert.Error(t, err)
```

Comment:
Nit, I personally prefer using `require` instead of `assert`, e.g. `require.Error(t, err)`. The problem with `assert` is that it doesn't immediately stop the test. And when there is a first error, usually that first error is followed by many following errors. This ends up cluttering the output, and in practice we usually only care about the first failure.

Reply:
4e09b0586f1084f24739b2952f7a77a5978072f7, only changed the asserts -> require for the files inside this PR, otherwise it's gonna be a lot of changes.